### ISSUE TITLE
License validation

### ISSUE THREAD

**Issue by Corniel** [dotnet-project-file-analyzers/dotnet-project-file-analyzers#337]
After releasing [Pro:0500: Only include packages with an explicitly defined license](https://dotnet-project-file-analyzers.github.io/rules/Proj0500.html) (#335 ) It might be worth to think about a strategy on how to check for valid licenses. After some consideration I think it might be worth to split it up in multiple rules.

## Proj0501: Only include packages with a compliant license
This rule ignores licenses that are not understood (license files/urls that could not be converted to a known license). Licenses that violate the licenses specified in the `<AllowedLicenses>` are guarded, otherwise an **ERROR** is added.

## Proj0502: Only include packages with a reviewed license
This rules ignores rules that are compliant with Proj0501. If the `<AllowedThirdPartyPackages>` has no entry, an **ERROR** is added. To prevent unintended license changes, I though it might be a good idea to add an hash based on de agreed on license:

``` XML
<AllowedThirdPartyPackages>
   SonarAnalyzer.CSharp #L9ThxnotKPzthJ7hu3bnORuT6xI,
   MyCompany.*          #EiVUfR3CPFNgdDHFhpG5OcrzIYY
</AllowedThirdPartyPackages>
```

Obviously, this implies that the hash is communicated in the error messages.. Furthermore, a hash mismatch should lead to another error message, and may even a different diagnostic code.

**Comment by CptWesley** [comment]
I was actually thinking about the same thing with the allowed list with hashes included to guard against licenses suddenly changing between versions. The only issue here might be the (already deprecated) license urls that point to a url where the content might change. I think it's better to just accept this edge case considering the urls are officially deprecated in the nuget spec anyway.

**Comment by Corniel** [comment]
Now I think of it - thanks for triggering it - we also need a rule to report on packages that only have an (unsolvable) `<licenseUrl>`. Something like: this package only uses a deprecated licenseUrl, please contact the maker to update the license.
